util/readpass_file.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "insecure_memzero.h"
+#include "warnp.h"
+
+#include "readpass.h"
+
+/* Maximum file length. */
+#define MAXPASSLEN 2048
+
+/**
+ * readpass_file(passwd, filename):
+ * Read a passphrase from ${filename} and return it as a malloced
+ * NUL-terminated string via ${passwd}.  Produce an error if the file is 2048
+ * characters or more, or if it contains any newline \n or \r\n characters
+ * other than at the end of the file.  Do not include the \n or \r\n
+ * characters in the passphrase.
+ */
+int
+readpass_file(char ** passwd, const char * filename)
+{
+	FILE * fp;
+	char passbuf[MAXPASSLEN];
+	char secondline[2];	/* We need 2 chars to handle EOLs. */
+
+	/* Open the file. */
+	if ((fp = fopen(filename, "r")) == NULL) {
+		warnp("fopen(\"%s\")", filename);
+		goto err1;
+	}
+
+	/* Get a line from the file. */
+	if ((fgets(passbuf, MAXPASSLEN, fp)) == NULL) {
+		if (ferror(fp)) {
+			warnp("fread(\"%s\")", filename);
+			goto err2;
+		} else {
+			/* We have a 0-byte password. */
+			passbuf[0] = '\0';
+		}
+	}
+
+	/* Bail if there's a second line. */
+	if ((fgets(secondline, 2, fp)) != NULL) {
```

Comment:
Why not `fgetc`?  That way you don't need a second buffer.

Reply:
done